content/tutorials/history.textile

```diff
@@ -31,17 +31,67 @@ We will be using channel rules in this tutorial to ensure all channels in the @p
 
 You have now enabled history for all channels in the @persisted@ namespace i.e. any channel with a name matching the pattern @peristed:*@ will store published messages to disk.
 
-h2. Step 3 - Install Ably
+h2. 
+  default: Step 3 – Install Ably
+  android: Step 3 – Set up environment and install Ably
 
 blang[java].
+  To start using Ably in your Java application, you need to include the Ably Client library. We recommend that you include the latest client library via Gradle in your project's @gradle.build@ file.
+  
+  ```[groovy]
+  apply plugin: 'application'
+  mainClassName = 'io.ably.tutorial.Example'
+
+  repositories {
+    jcenter()
+      maven { url "https://raw.github.com/paddybyers/Java-WebSocket/mvn-repo/" }
+  }
+
+  dependencies {
+      compile 'io.ably:ably-java:0.8.4'
+  }
+  ```
+
+  After you add neccessary dependencies, you can import AblyRest class into your code and initialize it.
+
+  ```[java]
+  import io.ably.lib.rest.AblyRest;
+
+  public class Example {
+    private final static String API_KEY = "INSERT-YOUR-API-KEY-HERE"; 
+    /* RuntimeException will be thrown if API_KEY will not be set to a proper one */
+    static {
+        if (API_KEY.contains("INSERT")) {
```

Comment:
This message is only intended for the tutorial code itself so that if someone runs it reminds them to add the API key.  Why would you put that code plus "INSERT-YOUR-API-KEY-HERE" in this?  You can stick with the convention of `Ably::Rest.new(key: api_key)` in the code we show in this tutorial (the code that matters), and in the tutorial repo itself you would have this code.  The `if (API_KEY.contains("INSERT")) {` has nothing to do with helping a user understand how to go about doing this.